code/modules/antagonists/traitor/datum_traitor.dm

```diff
@@ -19,8 +19,13 @@
 	can_hijack = HIJACK_HIJACKER
 
 /datum/antagonist/traitor/on_gain()
-	if(owner.current && isAI(owner.current))
-		traitor_kind = TRAITOR_AI
+	if(owner.current)
+		if(iscyborgshell(owner.current))
+			var/mob/living/silicon/robot/shell/cyborg_shell = owner.current
+			if(cyborg_shell.deployed)
+				cyborg_shell.undeploy()
+		if(isAI(owner.current))
+			traitor_kind = TRAITOR_AI
```

Comment:
Is this a change to help with dynamic? Why are you doing this.

Reply:
This is so AIs in shells become a malfunctioning AI instead of a regular traitor.